extensions/wikia/MarkWikiAsClosed/controllers/MarkWikiAsClosedController.class.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+class MarkWikiAsClosedController extends WikiaController {
+
+    const WIKI_ID = 'wikiId';
+    const REASON = 'reason';
+
+    public function init() {
+        $this->response->setFormat( WikiaResponse::FORMAT_JSON );
+    }
+
+    public function markWikiAsClosed() {
+
+        $wikiId = $this->request->getVal(self::WIKI_ID);
+        $reason = $this->request->getVal(self::REASON);
+
+        if ( empty( $wikiId) || empty($reason)) {
+            throw new BadRequestException();
+        }
+
+        $res = WikiFactory::setPublicStatus( WikiFactory::CLOSE_ACTION, $wikiId, $reason );
+
+        if ( $res === WikiFactory::CLOSE_ACTION ) {
```

Comment:
I think you should log cases where `$res` is different than expected. Afaik this shouldn't happen in our use case, but it's better to have that information.